models/marts/fct_order_items.sql

```diff
@@ -33,17 +41,35 @@ final as (
     -- dimensions
         order_items.price,
         order_items.freight_value,
-        case when order_payments.payment_type is null then 'returned'
-            else orders.order_status end as order_status,
+        case 
+            when order_payments.payment_type is null then 'returned'
+            else orders.order_status 
+        end as order_status,
         coalesce(order_payments.payment_type, 'returned') as payment_type,
         coalesce(order_payments.payment_installments, 0) as payment_installments,
         coalesce(order_payments.payment_value, 0) as payment_amount,
         reviews.review_score,
+        timestamp_diff(orders.order_delivered_carrier_date, orders.ordered_at, day) as days_from_ordered_to_carrier,
+        timestamp_diff(orders.order_delivered_customer_date, orders.ordered_at, day) as days_from_ordered_to_delivered,
+        timestamp_diff(orders.order_delivered_customer_date, orders.order_delivered_carrier_date, day) as days_from_carrier_to_delivered,
+        case
+            when orders.order_status = 'delivered' 
+                and extract(day from orders.order_delivered_customer_date) = extract(day from orders.order_estimated_delivery_date) then 'on time'
```

Comment:
the extract(day from ...) is a bit verbose and repetitive; a shorter alternative would be date(...). https://docs.snowflake.com/en/sql-reference/functions/to_date.html

Small thing but would make it more readable

Reply:
makes sense, moved logic to date()